Source/Classes/PINRemoteImageDownloadQueue.m

```diff
@@ -0,0 +1,145 @@
+//
+//  PINRemoteImageDownloadQueue.m
+//  PINRemoteImage
+//
+//  Created by Garrett Moon on 3/1/17.
+//  Copyright © 2017 Pinterest. All rights reserved.
+//
+
+#import "PINRemoteImageDownloadQueue.h"
+
+#import "PINURLSessionManager.h"
+#import "PINRemoteLock.h"
+
+@interface PINRemoteImageDownloadQueue ()
+{
+    NSUInteger _runningOperationCount;
+    PINRemoteLock *_lock;
+    
+    NSMutableArray <NSURLSessionDataTask *> *_highPriorityQueuedOperations;
+    NSMutableArray <NSURLSessionDataTask *> *_defaultPriorityQueuedOperations;
+    NSMutableArray <NSURLSessionDataTask *> *_lowPriorityQueuedOperations;
+}
+
+@end
+
+@implementation PINRemoteImageDownloadQueue
+
+@synthesize maximumNumberOfOperations = _maximumNumberOfOperations;
+
+- (PINRemoteImageDownloadQueue *)initWithMaximumNumberOfOperations:(NSUInteger)maximumNumberOfOperations
+{
+    if (self = [super init]) {
+        _maximumNumberOfOperations = maximumNumberOfOperations;
+        
+        _lock = [[PINRemoteLock alloc] initWithName:@"PINRemoteImageDownloadQueue Lock"];
+        _highPriorityQueuedOperations = [[NSMutableArray alloc] init];
+        _defaultPriorityQueuedOperations = [[NSMutableArray alloc] init];
+        _lowPriorityQueuedOperations = [[NSMutableArray alloc] init];
+    }
+    return self;
+}
+
+- (NSUInteger)maximumNumberOfOperations
+{
+    [self lock];
+        NSUInteger maximumNumberOfOperations = _maximumNumberOfOperations;
```

Comment:
Seems like a good performance improvement, but I want to avoid premature optimization at this time. Happy to address if it comes up in traces once this is in use.